### ISSUE TITLE
Pojo getter/setter naming does not completly follow java bean convention

### ISSUE THREAD

**Issue by dmiorandi** [jOOQ/jOOQ#5354]
**DB FIELD NAME**: 
```java
n_civ_residenza
```

**JOOQ JAVA METHODS** (seems wrong to me the uppercased 'N')
```java
nCivResidenza
getNCivResidenza
setNCivResidenza
```
**JAVA BEANS STANDARD / i.e. ECLIPSE/NETBEANS GENERATED METHODS** (i.e. el espression in jave EE expect this)
```java
nCivResidenza
getnCivResidenza
setnCivResidenza
```

This post explain exactly the issue:

http://www.google.com/url?q=http%3A%2F%2Fstackoverflow.com%2Fquestions%2F2948083%2Fnaming-convention-for-getters-setters-in-java&sa=D&sntz=1&usg=AFQjCNHASBoL1MnwdBZoWlMW1JBLxN7qUg

and this is the link to pdf documentation

http://download.oracle.com/otn-pub/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/beans.101.pdf?AuthParam=1466490341_284f08e0cb4b888cc12116a68e9b3368

it says

> Thus when we extract a property or event name from the middle of an existing Java name, we
> normally convert the first character to lower case. However to support the occasional use of all
> upper-case names, we check if the first two characters of the name are both upper case and if
> so leave it alone. So for example,
> “FooBah” becomes “fooBah”
> “Z” becomes “z”
> “URL” becomes “URL”
> We provide a method Introspector.decapitalize which implements this conversion rule.

**Comment by lukaseder** [comment]
Thank you for following up on this. I agree with the following:

> Thus when we extract a property or event name from the middle of an existing Java name, we
normally convert the first character to lower case. [...]

But notice that this is talking about the *property* name, not the "derived name" used inside of getters and setters. The latter is specified in 8.3

> If we discover a matching pair of “get<PropertyName>” and “set<PropertyName>” methods
that take and return the same type, then we regard these methods as defining a read-write property
whose name will be “<propertyName>”

I don't see any mention of a special case when `get<propertyName>` should be applied instead of `get<PropertyName>`. I think that Eclipse might just be a bit too clever here and apply some "magic" rule that also works for things like URL (translated to url), but I'm not so sure if there's really a bug in jOOQ here. Might be worth asking on Stack Overflow? :)

**Comment by lukaseder** [comment]
I may have been a bit too quick with my assumptions given your link on the user group towards this Stack Overflow question:
http://stackoverflow.com/a/16146215/521799

I guess the rule "all other tools do it the other way round" is sufficient here :)

**Comment by dmiorandi** [comment]
AFAIK also Netbeans and IntelliJ do the same as Eclipse. More than this, in my experience in javaEE (i.e. zkoss framework /ZUL binding in my case) when infering getter/setter by property name use same Eclipse naming.

So you are agree with me?
Anyway, actually I've got a work around  on generation so I'm ok.
I'm wondering, just if jooq will generate right syntax in the future with a breaking changes (new behavior) or with an optional configuration behavior (LEGACYNAMING / NEWONE) ore something like this.

**Comment by lukaseder** [comment]
> So you are agree with me?

Yes, I do now.

> LEGACYNAMING / NEWONE

There will be a "legacy naming" flag that is set to true in all patch releases that this gets merged into, and to false in the next minor release 3.9. So, new users will get the correct behaviour whereas old users can explicitly revert to the previous behaviour.

**Comment by lukaseder** [comment]
N.B: The `DefaultRecordMapper` also needs to be adapted. It will probably just support both types of naming schemes.

**Comment by dmiorandi** [comment]
excellent!

Tks Lukas

**Comment by dmiorandi** [comment]
Hi @lukaseder, any news about this? Do you think it could be really inserted in 3.10.0?
Tks

**Comment by lukaseder** [comment]
Hi @dmiorandi. Thanks for your reminder. Yes, I think this should be reasonable for jOOQ 3.10. Having two naming schemes supported in parallel, there's no risk for introducing a backwards incompatibility (although I disagree with myself from June 21: https://github.com/jOOQ/jOOQ/issues/5354#issuecomment-227415646, the new feature will be opt-in for compatibility reasons, as this implicit mapping change might lead to very hard to detect errors)

**Comment by lukaseder** [comment]
OK, this is now implemented in the code generator. Will implement the corresponding logic in the `DefaultRecordMapper` as well, soon: https://github.com/jOOQ/jOOQ/issues/3903 (definitely for jOOQ 3.10)

**Comment by dmiorandi** [comment]
Ok tks. Lukas.
